### PR TITLE
[scarb-doc] Add crate + initial CLI

### PR DESCRIPTION
**Pull request by drknzz**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -2,6 +2,7 @@
 members = [
     "scarb",
     "scarb-metadata",
+    "extensions/cairodoc",
```
Comment (maciektr) [inline Cargo.toml:5]:
```suggestion
    "extensions/scarb-doc",
```

File: extensions/cairodoc/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+name = "scarb-doc"
+version.workspace = true
+edition.workspace = true
+authors.workspace = true
+homepage.workspace = true
+license.workspace = true
+readme.workspace = true
+repository.workspace = true
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (maciektr) [inline extensions/cairodoc/Cargo.toml:11]:
```suggestion
```

**Review by maciektr — APPROVED** [review]

**Review by Arcticae — COMMENTED** [review]
Can we add a test? 🙏

File: extensions/scarb-doc/tests/test.rs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+use scarb_test_support::command::Scarb;
+
+#[test]
+fn test_main() {
+    let output = Scarb::quick_snapbox()
+        .arg("doc")
+        .arg("--crate-path")
+        .arg("tests/hello_world.cairo")
```
Comment (maciektr) [inline extensions/scarb-doc/tests/test.rs:8]: Consider enclosing the test data here, with use of https://docs.rs/assert_fs/latest/assert_fs/fixture/struct.TempDir.html

**Review by mkaput — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]

**Pull request merged**